jinja2/utils.py

```diff
@@ -284,6 +284,25 @@ def generate_lorem_ipsum(n=5, html=True, min=20, max=100):
     return Markup(u'\n'.join(u'<p>%s</p>' % escape(x) for x in result))
 
 
+def native_concat(invals):
+    '''
+    Jinja's native concat is just a list join. The list comes from result of
+    templating all the nodes. If the return type is expected to be
+    native, the list is artificial and we should return just the first
+    element.
+    '''
+    invals = [x for x in invals]
```

Comment:
I don't like building this list, seems wasteful of memory, although I guess there shouldn't be many nodes for the use case you're solving. Could change this to `head = list(islice(invals, 2))` to test the length, then either return `head[0]` or build a list from `invals` in `join`.